### PR TITLE
Add GraalVM native image metadata

### PR DESCRIPTION
**Pull request by morki**
This PR adds GraalVM native image metadata to enable building Mordant and Clikt with native-image tool.

Fixes regression in Clikt 4 https://github.com/ajalt/clikt/issues/415 

The only thing I have no experience in is adding those resources into library JAR, so please fix if `java-library` approach is not the right one.

### CODE REVIEW

**Comment by ajalt** [comment]
Awesome, thank you! It looks like only the linux metadata is included. Can you add the metadata for windows and macos, too? If not, I can do it after merge.

File: mordant/build.gradle.kts [hunk 1]
```diff
@@ -3,6 +3,7 @@
 plugins {
     kotlin("multiplatform")
     alias(libs.plugins.publish)
+    `java-library`
```
Comment (ajalt) [inline mordant/build.gradle.kts:6]:
```suggestion
```

No need for the java-library plugin, the kotlin plugin handles bundling the resources already.

**Comment by ajalt** [comment]
Thanks! I'll create the configs for other OSes. Did you do anything to generate these other than running an app with the native-image-agent?

File: mordant/src/jvmMain/resources/META-INF/native-image/com/github/ajalt/mordant/jni-config.json [hunk 2]
```diff
@@ -0,0 +1,594 @@
+[
+  {
+    "name": "[Z"
```
Comment (sschuberth) [inline mordant/src/jvmMain/resources/META-INF/native-image/com/github/ajalt/mordant/jni-config.json:3]: Sorry for the late question, but this entry looks odd, is it valid?

Comment (ajalt) [reply mordant/src/jvmMain/resources/META-INF/native-image/com/github/ajalt/mordant/jni-config.json:3]: I believe that's [a boolean array][1]. I don't know if it's necessary or not, but I don't thinks it's invalid.

[1]: https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-4.html#jvms-4.3.2

**Pull request merged**